packages/legend-studio/src/stores/editor-state/element-editor-state/service/ServiceRegistrationState.ts

```diff
@@ -59,40 +66,42 @@ interface ServiceVersionOption {
   value: Version | string;
 }
 
+export enum SERVICE_REGISTRATION_PHASE {
+  REGISTRATING_SERVICE = 'REGISTRATING_SERVICE',
+  ACTIVATING_SERVICE = 'ACTIVATING_SERVICE',
+}
+
 export class ServiceRegistrationState {
   editorStore: EditorStore;
   serviceEditorState: ServiceEditorState;
-  showModal = false;
   registrationState = ActionState.create();
   serviceEnv?: string | undefined;
   serviceExecutionMode?: ServiceExecutionMode | undefined;
   projectVersion?: Version | string | undefined;
   activatePostRegistration = true;
+  registrationPhase: SERVICE_REGISTRATION_PHASE | undefined;
```

Comment:
We don't need this. In `registrationState`, we can set the progress message. Let's use that instead